org.eclipse.scanning.api/src/org/eclipse/scanning/api/scan/ScanEstimator.java

```diff
@@ -175,48 +170,20 @@ public int getRank() {
 	/**
 	 * The estimated scan shape. Clue is in the name, ScanEstimator
 	 * @return
+	 * @throws GeneratorException 
 	 */
-	public int[] getShape() {
-		
-		if (shape!=null) return shape;
-	
-		Iterator<IPosition> iterator = generator.iterator();
-		IPosition last  = null;
-		int pointNum = 0;
-		long lastTime = System.currentTimeMillis();
-		while(iterator.hasNext()) {
-			last = iterator.next(); // Could be large...
-			pointNum++;
-			if (pointNum % 10000 == 0) {
-				long newTime = System.currentTimeMillis();
-				System.err.println("Point number " + pointNum++ + ", took " + (newTime - lastTime) + "ms");
+	public int[] getShape() throws GeneratorException {
+		if (shape == null) {
+			if (generator instanceof IPointGenerator<?>) {
+				shape =((IPointGenerator<?>) generator).getShape();
+			} else {
+				// can only get shape from IPointGenerator
+				// TODO: refactor ScanModel to have IPointGenerator instead of position iterable?
```

Comment:
Empty else block with a comment and a TODO in it seems wrong. Either you do something in the else and fix the TODO or remove the else and have a comment.